src/subscription_manager/scripts/rhsmcertd_worker.py

```diff
@@ -92,78 +99,165 @@ def _collect_cloud_info(cloud_list: list) -> dict:
 
         logger.info(f"Metadata and signature gathered for cloud provider: {cloud_provider_id}")
 
-        # Encode metadata and signature using base64 encoding. Because base64.b64encode
-        # returns values as bytes, then we decode it to string using ASCII encoding.
-        b64_metadata: str = base64.b64encode(bytes(metadata, "utf-8")).decode("ascii")
-        b64_signature: str = base64.b64encode(bytes(signature, "utf-8")).decode("ascii")
```

Comment:
You are right, this is a mistake with current mock implementation of adapters in Candlepin. This will be reverted back once bare Candlepin supports it.